src/importlinter/adapters/user_options.py

```diff
@@ -51,3 +61,37 @@ def _clean_section_config(section_config: Dict[str, Any]) -> Dict[str, Any]:
             else:
                 section_dict[key] = value.strip().split("\n")
         return section_dict
+
+
+class TomlUserOptionReader(ports.UserOptionReader):
+    """
+    Reader that looks for and parses the contents of INI files.
+    """
+
+    section_name = "importlinter"
+
+    def read_options(self, config_filename: Optional[str] = None) -> Optional[UserOptions]:
+        if not _HAS_TOML:
+            return None
+
+        if config_filename:
+            config_filenames = file_finding.find_any(config_filename)
+            if not config_filenames:
+                raise FileNotFoundError(f"Could not find {config_filename}.")
+        else:
+            config_filenames = file_finding.find_any("pyproject.toml")
+            if not config_filenames:
+                return None
+
+        try:
+            data = toml.load(config_filenames[0])
+        except toml.TomlDecodeError:
+            return None
```

Comment:
NIP: Maybe also some message should be logged on `DEBUG`, so users will notice that something is wrong with their `toml` configuration

Reply:
The project doesn't use a logger yet, therefore I'm not sure if it makes sense to just use logging.info/debug or if we need a proper logger setup

Reply:
That's true, so we can setup logger in separate PR

Reply:
ok great

Reply:
"Errors should never pass silently." I would lose the `try...except` and instead let the TomlDecodeError bubble up to be reported to the end user. This is [where that happens](https://github.com/seddonym/import-linter/blob/e2dff4f885954f9414a3e23420aa3e13c478aee7/src/importlinter/application/use_cases.py#L37-L41).

Reply:
Let's image we have another reader in the future, e.g. for JSON. Your suggestion would lead to an error if the user specified a JSON file over the CLI. The TOML parser will fail obviously and therefore block a further delegation. See https://github.com/seddonym/import-linter/pull/99#discussion_r676922797

Reply:
The problem I see is, that file extensions are optional. What about the following:
If no config file is given and the parsing of one default file fails. report the issue to the user. If the user is giving a file, don't report parsing errors.

The positive side of this is the quite easy implementation with e.g. a custom parsing error

Reply:
> If the user is giving a file, don't report parsing errors.

I'm not convinced. Currently Import Linter behaves like this:

```
$ lint-imports --config=invalidfile.ini
While reading from '<string>' [line  9]: section 'importlinter:contract:1' already exists 
```

It feels like a step backwards to change it to this:
```
$ lint-imports --config=invalidfile.ini
Could not read any configuration.
```
Something like this would be much better:

```
$ lint-imports --config=invalidfile.ini
Malformed INI file:
While reading from '<string>' [line  9]: section 'importlinter:contract:1' already exists 
```

As you can see it is similar to before, but displays the detected file format first.

> The problem I see is, that file extensions are optional.

For backwards compatibility, we could just default to INI:

```
$ lint-imports --config=invalidfile.txt
Malformed INI file:
While reading from '<string>' [line  9]: section 'importlinter:contract:1' already exists 
```

What do you think?

Reply:
So, in summary, if a user provides a file:
--> toml file extension --> use toml
--> everything else --> use ini
Correct?

If there is a need to have toml files w/o toml extensions we can add the support later for this feature

Reply:
Exactly!

> If there is a need to have toml files w/o toml extensions we can add the support later for this feature

Agreed.

Reply:
nice, let me add this

Reply:
I added it with some changes to the reader logic. Please review

Reply:
Looks good, but could do with test coverage of the new behaviour. If you're able to do this, great! If not then I will add it at the same time as tackle the functional tests.